extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -46,6 +47,11 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 	public currentBook: BookModel;
 	supportedTypes = ['text/treeitems'];
 
+	public undoTocFiles: any[] = [];
+	public undoMovedFiles: Map<string, string>[] = [];
+	public redoTocFiles: any[] = [];
```

Comment:
We should probably put a limit on the max stored undo/redo operations so it doesn't grow infinitely large if they're moving lots of stuff around.